src/components/network-chart/index.js

```diff
@@ -60,74 +63,127 @@ class NetworkChart extends React.PureComponent<Props> {
+      width: markerLength + '%',
+      left: markerPosition + '%',
+    };
+    return markerStyling;
+  }
+
+  // Create row with correct details
+  _renderRow(nodeId: any, index: number) {
+    const markersPerRange = this._filterMarkersPerRange(
+      this.props.timeRange,
+      this.props.markers
+    );
+
+    const itemClassName = (
+      'item ' + markersPerRange[index].status
+    ).toLowerCase();
+
+    return (
+      <section className={itemClassName}>
+        <div className="status" />
+        <div className="itemLabel">
+          {markersPerRange[index].URI} - {markersPerRange[index].dur}
+        </div>
+        <div
+          className="itemBar"
+          style={this._getMarkerStyling(markersPerRange[index])}
+        >
+          &nbsp;
+        </div>
+      </section>
+    );
+  }
+
+  _filterMarkersPerRange(timeRange: Object, markers: []) {
+    return markers.filter(function(marker) {
+      const updateRange =
+        timeRange.start < marker.startTime && timeRange.end > marker.endTime;
+      return updateRange;
+    });
+  }
+
   render() {
-    const {
-      maxNetworkRows,
-      timeRange,
-      threadIndex,
-      networkTimingRows,
-      markers,
-      previewSelection,
-      updatePreviewSelection,
-    } = this.props;
+    const { timeRange, markers } = this.props;
 
-    if (!networkTimingRows.length) {
+    //eslint-disable-next-line
+    const networkMarkersPerRange = this._filterMarkersPerRange( timeRange, markers );
+    if (!networkMarkersPerRange.length) {
       return <NetworkChartEmptyReasons />;
     }
 
     // The viewport needs to know about the height of what it's drawing, calculate
     // that here at the top level component.
-    const maxViewportHeight = maxNetworkRows * ROW_HEIGHT;
+    // const maxViewportHeight = maxNetworkRows * ROW_HEIGHT;
 
     return (
       <div className="networkChart">
-        <MarkerChartCanvas
-          key={threadIndex}
-          viewportProps={{
-            timeRange,
-            previewSelection,
-            maxViewportHeight,
-            viewportNeedsUpdate,
-            maximumZoom: this.getMaximumZoom(),
-          }}
-          chartProps={{
-            networkTimingRows,
-            markers,
-            updatePreviewSelection,
-            rangeStart: timeRange.start,
-            rangeEnd: timeRange.end,
-            rowHeight: ROW_HEIGHT,
-            threadIndex,
-          }}
+        <VirtualList
+          className="treeViewBody"
+          items={networkMarkersPerRange}
+          renderItem={this._renderRow}
+          itemHeight={ROW_HEIGHT}
+          columnCount={1}
+          focusable={true}
+          onKeyDown={false}
+          specialItems={[]}
+          disableOverscan={false}
+          onCopy={false}
+          getHoveredItemInfo={this.getHoveredMarkerInfo}
+          // If there is a deep call node depth, expand the width, or else keep it
+          // at 3000 wide.
+          containerWidth={3000}
```

Comment:
The container width needs to be dynamic. Shall this work the same way as in the call tree?
@julienw